projects/packages/account-protection/src/class-password-manager.php

```diff
@@ -95,26 +95,33 @@ public function validate_password_reset( \WP_Error $errors, $user ): void {
 	/**
 	 * Handle the profile update.
 	 *
-	 * @param int      $user_id The user ID.
-	 * @param \WP_User $old_user_data Object containing user data prior to update.
+	 * @param int                $user_id The user ID.
+	 * @param \WP_User|\stdClass $old_user_data Object containing user data prior to update.
 	 *
 	 * @return void
 	 */
-	public function on_profile_update( int $user_id, \WP_User $old_user_data ): void {
+	public function on_profile_update( int $user_id, $old_user_data ): void {
 		// phpcs:ignore WordPress.Security.NonceVerification
 		if ( isset( $_POST['action'] ) && $_POST['action'] === 'update' ) {
-			$this->save_recent_password_hash( $user_id, $old_user_data->user_pass );
+			// Only save if user_pass property exists and is not empty
+			if ( isset( $old_user_data->user_pass ) && '' !== $old_user_data->user_pass ) {
```

Comment:
```suggestion
			if ( isset( $old_user_data->user_pass ?? null ) && '' !== $old_user_data->user_pass ) {
```

We've relaxed the type checking for the `old_user_data` argument, let's stay on the safe side :-)